---
title: Difference between fibroblasts with bclxl treatment and fibroblasts with onecut2
  treatment
author: "Dian Dupon & Thijmen Weijgertze"
date: '`r Sys.Date()`'
output: 
  html_document:
    toc: true
    toc_float: true
    theme: lumen
---
<div style="text-align: justify"> 

# Introductie

Door middel van transdifferentiatie kan een cel differentiëren tot een ander celtype zonder eerst te differentiëren naar een pluripotentestamcel. Bij transdifferentiatie wordt een transcriptie factor tot overexpressie gebracht, waardoor de cel differentiëerd in een ander celtype. Deze methode zou veel tijd schelen in het differentiëren van cellen. De transcriptiefactor ONECUT2 (OC-2) zou bij overexpressie mogelijk fibroblasten kunnen laten differiënteren tot hersencellen. In dit onderzoek werd er onderzocht wat de verschillen in genexpressie zijn tussen BCLXL (controle transcriptie factor) behandelde cellen en ONECUT2 behandelde cellen.

# Packages laden + paden opslaan

```{r packages, warning=FALSE, message=FALSE}
# inladen alle benodigde packages
library(tidyverse)
library(here)
library(DESeq2)
library(pheatmap)
library(org.Hs.eg.db)
library(GOstats)
library(png)
library(grid)
library(gridExtra)
```

```{r opslaan data in variabelen, warning=FALSE, message=FALSE}
# opslaan van data en r objecten in variabelen
metadata_OC2 <- read.csv("/home/daur2/rnaseq/rnaseq_onecut/onecut_sampledata_OC2.csv")
read_counts_OC2 <- readRDS("/home/daur2/rnaseq/rnaseq_onecut/counts/read_counts_OC2.rds")

# opslaan van paden in variabelen
fastq_output_path <- "/home/daur2/rnaseq/rnaseq_onecut/fastq_output/"
fastq_path <- "/home/daur2/rnaseq/rnaseq_onecut/fastq/"
bam_path <- "/home/daur2/rnaseq/rnaseq_onecut/bam/"
images_path <- "~/daur2/bclxl_onecut2_project/images/"
data_path <- "~/daur2/bclxl_onecut2_project/data/"
```

# Genereren van de Counttable

```
# Create object with input dir for bam files
bam_dir <- "/home/daur2/rnaseq/rnaseq_onecut/bam/"

# Create object with output dir for count tables
counts_dir <- "/home/daur2/rnaseq/rnaseq_onecut/counts/"

# Create vector with names of the correct bam files
bam_files <- list.files(bam_dir, pattern = ".*[0349]\\.bam$", full.names = TRUE)

# Count the reads per gene using the in-built NCBI RefSeq annotations
read_counts <- featureCounts(
  
  files = bam_files,
  annot.inbuilt = "hg38",
  useMetaFeatures = TRUE,
  strandSpecific = 0,
  isPairedEnd = TRUE, 
  countReadPairs = TRUE, 
  nthreads = 10
)
```

# Creëren van een DESeq object

```{r DESeq object, warning=FALSE, message=FALSE}
# opslaan count matrix
count_matrix <- read_counts_OC2$counts

# class naar dataframe veranderen
metadata_OC2 <- as.data.frame(metadata_OC2)

# rownames toevoegen aan metadata_OC2
rownames(metadata_OC2) <- paste0(metadata_OC2$Run, ".bam")

# controleren of de col names van de count table gelijk zijn aan de rownames van metadata_OC2
colnames(count_matrix) == rownames(metadata_OC2) 

# Een factor maken van celltype
metadata_OC2$Cell_type <- metadata_OC2$Cell_type %>% 
  factor(levels = c("Skin derived fibroblast overexpressing Bclxl", "2 days after induction of OC2 in skin derived fibroblasts"))

# DESeq2 object aanmaken
dds <- DESeqDataSetFromMatrix(
  countData = count_matrix,
  colData = metadata_OC2,
  design = ~ Cell_type
)
```

# PCA analyse

```{r normalizing and PCA analysis, warning=FALSE, message=FALSE, fig.cap="***Figuur 3; PC percentage staafdiagram,*** In het figuur staan de percentages van de PC summary"}
# normaliseren van de data door geoptimaliseerde log2 normalisatie
dds_normalized <- rlog(dds)

# uitvoeren PCA op iPSC study
pca <- dds_normalized %>% assay() %>% t() %>% prcomp()

# Het vinden van de proportie van variatie voor elke PC
pca_summary <- summary(pca)$importance
pca_summary

# Creëren van een data frame met metadata en de PCA waarden voor het plotten
pca_plotting <- cbind(metadata_OC2, pca$x)

# Het verkrijgen van de variatiepercentages die gedekt worden door PC1 en PC2
PC1_var <- round(pca_summary["Proportion of Variance", "PC1"]*100, digits = 1)
PC2_var <- round(pca_summary["Proportion of Variance", "PC2"]*100, digits = 1)
PC3_var <- round(pca_summary["Proportion of Variance", "PC3"]*100, digits = 1)
PC4_var <- round(pca_summary["Proportion of Variance", "PC4"]*100, digits = 1)
```

```{r PC staafdiagram, warning=FALSE, message=FALSE, fig.cap="***Figuur 3; PC percentage staafdiagram,*** In het figuur staan de percentages van de PC summary"}
# tibble aanmaken met de PC variablelen
PC_percentages_table <- tibble(PC1_var, PC2_var, PC3_var, PC4_var)

#tidy maken van de tibble
PC_percentages_table <- pivot_longer(PC_percentages_table, cols = c(colnames(PC_percentages_table)), names_to = "PC", values_to = "percentage")
PC_percentages_table$PC <- c("PC1", "PC2", "PC3", "PC4")
head(PC_percentages_table)

# PC percentages in een staafdiagram plotten
PC_percentages_table %>%
  ggplot(aes(x=PC, y=percentage))+
  geom_col()+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 60, hjust = 1))+
  coord_cartesian (ylim =c(0,100))+
  theme(legend.position="none")+
  labs(
    title = "PC variation percentages",
    x = "",
    y = "PC percentage (%)"
  )
```

Uit het figuur valt te concluderen dat de meeste variatie zicht in bevond in PC1 (94%), in PC2 bevond zich 3% van de variatie, in PC3 2% van de variatie en in PC4 geen variatie.

```{r PCA plot, warning=FALSE, message=FALSE, fig.cap="***Figuur 3; PCA plot van het onecut2 onderzoek,*** In het figuur staat PC1 tegenover PC2. De data is gegroepeerd op zowel celltype. Alle celltypen zijn afkomstig cellijn: Fibroblast line 1 (CL1500023)"}
ggplot(pca_plotting) + 
  geom_point(aes(x=PC1, y=PC2, color = Cell_type, shape = Cell_line), size = 5) +
  ggtitle("PCA voor het onecut2 onderzoek") +
  xlab(paste0("PC1 (", PC1_var, "%)")) +
  ylab(paste0("PC2 (", PC2_var, "%)")) +
  theme_bw()
```

Uit figuur 3 isvolgens de PC1 variatie gebleken dat de data gegroepeerd is in twee groepen corresponderend met de condities "Skin derived fibroblast overexpressing Bclxl" en "2 days after induction of OC2 in skin derived fibroblasts". PC2 liet zien dat ... (nog bespreken)

# Uitvoeren DGE analyse

```{r dge analyse, warning=FALSE, message=FALSE}
dge_analyse <- DESeq(dds)

# Obtain the results for the DGE analysis
# default settings
dge_results <- results(dge_analyse)

# lfc > 1 alpha 0.01
dge_results_lfc1_alpha01 <- results(dge_analyse, lfcThreshold = 1, alpha = 0.01)
```

# Volcano plot opstellen

```{r volcano plot, warning=FALSE, message=FALSE, fig.cap="***Figuur 3; Volcano van het onecut2 onderzoek,*** In het figuur staat de log2 foldchange op X-as tegenover de adjusted P waarde op de Y-as. Met een padj kleiner dan 0.01 en een log2 foldchange hoger dan 1 of -1 is oranjegemarkeerd"}
# Create a dataframe for plotting without genes with padj = NA
sign_upregulated_genes <- data.frame(dge_results_lfc1_alpha01) %>% filter(!is.na(padj), padj < 0.01, log2FoldChange > 1) %>% nrow()
sign_upregulated_genes
sign_downregulated_genes <- data.frame(dge_results_lfc1_alpha01) %>% filter(!is.na(padj), padj < 0.01, log2FoldChange < -1) %>% nrow() 
sign_downregulated_genes

dge_plotting <- data.frame(dge_results) %>% filter(!is.na(padj))

# Create column specifying if gene is significantly differentially expressed
dge_plotting <- dge_plotting %>% 
  mutate(signif = if_else(padj < 0.01 & abs(log2FoldChange) > 1, 
                          "Significant", "Not significant"))

dge_plotting %>% filter(padj < 0.01 & abs(log2FoldChange) > 1)

# Create a volcano plot
dge_plotting %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(padj), color = signif)) +
  geom_point() + 
  xlab("log2 fold change") +
  ylab("-log10 adjusted p-value") + 
  theme_bw() +
  
  # Change the legend text and the colours
  scale_colour_manual(values = c("grey", "darkorange"), name = "Significance") +
  
  annotate("text", x = -5, y = 250, 
           label = paste("Downregulated genes: \n", sign_downregulated_genes), colour = "black")+
  annotate("text", x = 10, y = 250, 
           label = paste("Upregulated genes: \n", sign_upregulated_genes), colour = "black")+
  
  # threshold bepalen door stippellijn
  geom_hline(yintercept = -log10(0.01), linetype = "dashed")+
  geom_vline(xintercept = 1, linetype = "dashed")+
  geom_vline(xintercept = -1, linetype = "dashed")
```

Uit de vulcona plot is geconcludeerd dat een zeer groot aantal genen een significant verschil in expressie laten zien met een LFC hoger dan 1. Verder is er geconcludeerd dat er meer significante genen zijn waarbij sprake is van upregulatie van de genexpressie ten opzichte van downregulatie.

# Pheatmap opstellen

```{r pheatmap maar met symbols, warning=FALSE, message=FALSE}
# Obtain the gene identifiers of the 15 most significant genes
dge_results_filtered <- dge_results[which(dge_results$padj < 0.01),]

top_5_up <- dge_results_filtered[order(dge_results_filtered$log2FoldChange, decreasing = TRUE), ] 
top_5_up <- head(top_5_up, n=5)
top_5_down <- dge_results_filtered[order(dge_results_filtered$log2FoldChange, decreasing = FALSE), ] 
top_5_down <- head(top_5_down, n=5)

top10_genes <- rbind(top_5_up, top_5_down)
top10_genes

top10_genes <- rownames(top10_genes) %>% data_frame()
colnames(top10_genes)[1] <- "entrezid"

# Obtain the count values for these genes
count_values <- assay(dds)[top10_genes$entrezid,]
count_values 

# Obtain the gene symbols based on the Entrez identifiers
rownames(count_values) <- mapIds(org.Hs.eg.db,
                             keys = rownames(count_values),
                             column = "SYMBOL",
                             keytype = "ENTREZID",
                             multiVals = "first")

# Replace the column names with conditions
colnames(count_values) <- colData(dds)$cell_type
count_values

# Visualize the results in a heatmap (with scaling!) 
pheatmap(count_values, scale = "row", show_rownames = TRUE)
```

Uit de pheatmap is geconcludeerd dat ONECUT2 zich bevond tot de top 5 upgereguleerde genen en daadwerkelijk door de transcriptiefactor OC-2 upgereguleerd is.

# Functie opstellen voor het converten van entrezid's naar uniprot, ensemble en symbolen

```{r}
EntrezConverter <- function(entrezid){
  
stopifnot(is.character(entrezid))

entrezConverterData <- dge_results %>% data.frame()
entrezConverterData <- entrezConverterData %>% mutate(ENTREZID = rownames(entrezConverterData))

# Find the associated GO terms
entrezConverterData$UNIPROT <- mapIds(org.Hs.eg.db,
                keys = entrezConverterData$ENTREZID,
                column = "UNIPROT",
                keytype = "ENTREZID",
                multiVals = "first")
entrezConverterData$ENSEMBL <- mapIds(org.Hs.eg.db,
                keys = entrezConverterData$ENTREZID,
                column = "ENSEMBL",
                keytype = "ENTREZID",
                multiVals = "first")
entrezConverterData$SYMBOL <- mapIds(org.Hs.eg.db,
                keys = entrezConverterData$ENTREZID,
                column = "SYMBOL",
                keytype = "ENTREZID",
                multiVals = "first")

# Find the associated GO terms
entrezConverterResults <- entrezConverterData %>% filter(ENTREZID %in% entrezid)
entrezConverterResults <- entrezConverterResults[c(7, 10:8)] 
entrezConverterResults <- as_tibble(entrezConverterResults)
entrezConverterResults
}

EntrezConverter(entrezid = c("3175", "9480", "390874"))
```

# GOterm analyse

```{r}
gotermAnalysis <- function(x, upregulated, lfc_limiter, padj_limiter){

# stops the script in case of a wrong output  
stopifnot(is.logical(upregulated))
stopifnot(is.numeric(lfc_limiter))
stopifnot(is.numeric(padj_limiter))
stopifnot(padj_limiter > 0)

# dge results opslaan in var
dge_results <- x

# if statement voor de upregulated logical 

if(upregulated == TRUE){

# Create a list of upregulated genes
upregulated_genes <- dge_results %>% data.frame() %>% filter(log2FoldChange > lfc_limiter, padj < padj_limiter) %>% rownames()
up_or_downregulated_genes <- upregulated_genes

}else if(upregulated == FALSE)
{
  
# Create a list of downregulated genes
downregulated_genes <- dge_results %>% data.frame() %>% filter(log2FoldChange < -(lfc_limiter), padj < padj_limiter) %>% rownames()
up_or_downregulated_genes <- downregulated_genes

}

# Create a list of all genes in the dataset
all_genes <- dge_results %>% data.frame() %>% rownames()

# Perform GO term enrichment analysis
test_object <- new("GOHyperGParams",
                   geneIds = up_or_downregulated_genes,
                   universeGeneIds = all_genes, 
                   annotation = "org.Hs.eg.db", 
                   ontology = "BP", 
                   pvalueCutoff = 1,
                   testDirection = "over")
summary(hyperGTest(test_object))
} 
```


```{r upregulated goterm analysis graph, message=FALSE, warning=TRUE}
### UPREGULATED GRAPH
# opslaan goterm Analyse upregulated
goterm_analysis_upregulated_results <- gotermAnalysis(
  dge_results, 
  upregulated = TRUE, 
  lfc_limiter = 1, 
  padj_limiter = 0.01)

# Adjust the p values for multiple testing
goterm_analysis_upregulated_results$padj <- p.adjust(goterm_analysis_upregulated_results$Pvalue, method = "BH")

# Select only gene sets that are larger than 5 but smaller 
# than 500 (to prevent taking along very small and very large
# gene sets).
goterm_analysis_upregulated_results <- goterm_analysis_upregulated_results %>% filter(Count > 5) %>% filter(Count < 500)

# Select the top 20 GO terms
goterm_analysis_up_top20 <- goterm_analysis_upregulated_results[order(goterm_analysis_upregulated_results$padj)[1:20],]

# Plot the p-values of the top 20 GO terms
goterm_analysis_up_top20$Term <- factor(goterm_analysis_up_top20$Term, 
                                     levels = goterm_analysis_up_top20$Term[
                                       order(goterm_analysis_up_top20$padj, decreasing = TRUE)])
goterm_analysis_up_top20 %>% ggplot(aes(x = Term, y = -log10(padj))) +
  geom_point() +
  coord_flip() +
  ylab(expression(-log[10](adjusted~italic(P)~value))) + 
  xlab("GO terms") +
  ggtitle("Top 20 enriched GO terms\n for upregulated genes") +
  theme_bw()
```

```{r downregulated goterm analysis graph, warning=FALSE, message=FALSE}
### DOWNREGULATED GRAPH
# opslaan goterm Analyse downregulated
goterm_analysis_downregulated_results <- gotermAnalysis(
  dge_results, 
  upregulated = FALSE, 
  lfc_limiter = 1, 
  padj_limiter = 0.01)

# Adjust the p values for multiple testing
goterm_analysis_downregulated_results$padj <- p.adjust(goterm_analysis_downregulated_results$Pvalue, method = "BH")

# Select only gene sets that are larger than 5 but smaller 
# than 500 (to prevent taking along very small and very large
# gene sets).
goterm_analysis_downregulated_results <- goterm_analysis_downregulated_results %>% filter(Count > 5) %>% filter(Count < 500)

# Select the top 20 GO terms
goterm_analysis_down_top20 <- goterm_analysis_downregulated_results[order(goterm_analysis_downregulated_results$padj)[1:20],]

# Plot the p-values of the top 20 GO terms
goterm_analysis_down_top20$Term <- factor(goterm_analysis_down_top20$Term, 
                                     levels = goterm_analysis_down_top20$Term[
                                       order(goterm_analysis_down_top20$padj, decreasing = TRUE)])
goterm_analysis_down_top20 %>% ggplot(aes(x = Term, y = -log10(padj))) +
  geom_point() +
  coord_flip() +
  ylab(expression(-log[10](adjusted~italic(P)~value))) + 
  xlab("GO terms") +
  ggtitle("Top 20 enriched GO terms\n for downregulated genes") +
  theme_bw()
```


Door de onderzoekers werd vermoed dat transdifferentiatie met ONECUT2 resulteert in de differentiatie van fibroblasten in hersencellen. Onder invloed van transcriptiefactor ONECUT2 (gedurende 2 uur) worden er genen anders gereguleerd ten opzichte van fibroblasten met alleen het transcriptiefactor BCLXL. Het viel op dat genen met betrekking tot de hersenen en de eventuele differentiatie naar hersencellen upgereguleerd of downgereguleerd waren. Dit waren genen binnen GOterms als "neurogenesis", “generation of neurons”, “neuron differentiation”, “synaptic signaling” en “nervous system development”. Aan de hand van deze resultaten kon geconcludeerd worden dat het eerder genoemde vermoeden van de onderzoekers klopt.

</div>