---
title: "Formatieve opdracht"
author: "Thijmen Weijgertze"
date: "4/3/2023"
output: html_document
---

# Species identification and abundance estimation

### FastQC analysis

```{r, echo=TRUE, eval=FALSE}
# update conda and answer y to proceed question
echo “y” | conda update -n base conda
```

```{bash, engine.opts='-i', echo=TRUE, eval=TRUE}
# show all conda virtual environments
conda env list

# show yml file content
cat ~/daur2/metagenomics/yml/setup_meta_env.yml
```

```{bash, engine.opts='-i', echo=TRUE, eval=FALSE}
# create conda virtual environment
conda env create --file ~/daur2/metagenomics/yml/setup_meta_env.yml
```

```{bash, engine.opts='-i', echo=TRUE, eval=FALSE}
# activate environment
conda activate meta

# install fastqc and answer yes for procceed question
echo "y" | conda install -c bioconda fastqc

# deactivate conda virtual environment
conda deactivate
```

```{bash, engine.opts='-i', echo=TRUE, eval=FALSE}
# activate environment
conda activate meta

# perform FastQC analysis
fastqc -o ~/daur2/metagenomics/formatieve_opdracht/fastqc/ /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R1_001.fastq.gz
fastqc -o ~/daur2/metagenomics/formatieve_opdracht/fastqc/ /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R2_001.fastq.gz

# deactivate conda virtual environment
conda deactivate
```

```{r quality per base images, echo=FALSE, warning=FALSE, message=FALSE, fig.height=3, out.width= '100%', fig.cap="Fig. 1: Quality distribution: Left-side) Foward read quality scores range from 0 to 40 on the Phred scale. All bases for all reads have very high quality (>30), indicating that our reads have an accuracy of >99,9%. Right-side) reversed read quality scores range from 0 to 40 on the Phred scale. All bases for all reads have very high quality (>30). This indicates our reads have an accuracy of >99,9%"}
library(here)
library(png)
library(grid)
library(gridExtra)

img1 <-  rasterGrob(as.raster(readPNG(here("metagenomics/formatieve_opdracht/images/fastqc_mock2_R1_per_base_quality.png"))))
img2 <-  rasterGrob(as.raster(readPNG(here("metagenomics/formatieve_opdracht/images/fastqc_mock2_R2_per_base_quality.png"))))
grid.arrange(img1, img2, ncol=2, top=textGrob("Per base quality distribution of forward (right) and reverse (left) reads", gp=gpar(fontsize=10,font=8)))
```

### Species identification using minikraken

```{bash installing kraken, echo=TRUE, eval=FALSE}
# activate environment
conda activate meta

#install kraken2 and answer y to proceed question
echo "y" | conda install -c bioconda kraken2

conda deactivate
```

```{bash, echo=TRUE, eval=FALSE}
# activate environment
conda activate meta

kraken2 --db /home/daur2/metagenomics/minikraken2_v2_8GB_201904_UPDATE/ --threads 2 --paired --gzip-compressed --output ~/daur2/metagenomics/formatieve_opdracht/kraken2/mock2.kraken --report ~/daur2/metagenomics/formatieve_opdracht/kraken2/mock2.report --use-names /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R1_001.fastq.gz /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R2_001.fastq.gz

conda deactivate
```

Kraken2 identification statistics:

- Classified: 59541308 sequences classified (90.33%)
- Unclassified: 6374528 sequences unclassified (9.67%)
- Threads: 1
- Duration: 65915836 sequences (19439.90 Mbp) processed in 1734.357s (2280.4 Kseq/m, 672.52 Mbp/m)
- .kraken size: 13.3 GB
- .report size: 272 KB

```{bash, echo=TRUE, eval=FALSE}
# activate environment
conda activate meta

# installing bioconda bracken
echo "y" | conda install -c bioconda bracken

conda deactivate
```

```{bash, echo=TRUE, eval=FALSE}
# activate environment
conda activate meta

# bracken
bracken -d /home/daur2/metagenomics/minikraken2_v2_8GB_201904_UPDATE/ -i ~/daur2/metagenomics/formatieve_opdracht/kraken2/mock2.report -o ~/daur2/metagenomics/formatieve_opdracht/bracken/mock2.bracken

conda deactivate
```

```{bash, echo=TRUE, eval=FALSE}
# activate environment
conda activate kraken_biom_v101

# install kraken-biom and answer y to proceed question
echo "y" | conda install -c "bioconda/label/cf201901" kraken-biom

conda deactivate
```

```{bash, echo=TRUE, eval=FALSE}
# activate environment
conda activate kraken_biom_v101

kraken-biom ~/daur2/metagenomics/formatieve_opdracht/kraken2/mock2_bracken_species.report --fmt json -o ~/daur2/metagenomics/formatieve_opdracht/kraken2/mock2_bracken_species.biom

conda deactivate
```

Number of species in sample: 1705 

### visualisation of data

```{r, echo=TRUE, eval=FALSE}
# load nessecary packages 
BiocManager::install("phyloseq")
```

```{r, echo=TRUE, eval=TRUE}
library(phyloseq)
library(ggplot2)
data <- "~/daur2/metagenomics/formatieve_opdracht/kraken2/mock2_bracken_species.biom"

merged_metagenomes <- import_biom(data)
#View(merged_metagenomes@tax_table@.Data)
```

```{r, echo=TRUE, eval=TRUE}
# remove taxonomy information from values/ remove first four characters
merged_metagenomes@tax_table@.Data <-
  substring(merged_metagenomes@tax_table@.Data, 4)

# rename column header to informative format
colnames(merged_metagenomes@tax_table@.Data) <- 
  c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

# inspect data format
# View(merged_metagenomes@tax_table@.Data)

# using a function subset_taxa() we can separate the bacterial data and store it in a separate variable:
merged_metagenomes_b <- subset_taxa(merged_metagenomes, Kingdom == "Bacteria")

# then using a function called sample_sums() we can count how many reads are classified as bacterial:
sample_sums(merged_metagenomes_b)

# to provide informative headers use the function sample_names():
sample_names(merged_metagenomes_b) <- "bacteria"
```

```{r, echo=TRUE, eval=TRUE}
# retrieve the present taxonomic kingdoms from our data
unique(merged_metagenomes@tax_table@.Data[,"Kingdom"])

# subset kingdoms
merged_metagenomes_b <- subset_taxa(merged_metagenomes, Kingdom == "Bacteria")
merged_metagenomes_e <- subset_taxa(merged_metagenomes, Kingdom == "Eukaryota")
merged_metagenomes_a <- subset_taxa(merged_metagenomes, Kingdom == "Archaea")
merged_metagenomes_v <- subset_taxa(merged_metagenomes, Kingdom == "Viruses")

# rename samples
sample_names(merged_metagenomes_b) <- "bacteria"
sample_names(merged_metagenomes_e) <- "eukaryote"
sample_names(merged_metagenomes_a) <- "archea"
sample_names(merged_metagenomes_v) <- "virus"

# count number of reads in total
sample_sums(merged_metagenomes)

# count number of reads per kingdom
c(sample_sums(merged_metagenomes_b), sample_sums(merged_metagenomes_e), sample_sums(merged_metagenomes_a), sample_sums(merged_metagenomes_v))
```

```{r, echo=TRUE, eval=TRUE}
# collect kingdoms seperately in data frames
data_b <- data.frame(Samples = sample_names(merged_metagenomes_b),
                   Reads = sample_sums(merged_metagenomes_b))

data_e <- data.frame(Samples = sample_names(merged_metagenomes_e),
                   Reads = sample_sums(merged_metagenomes_e))

data_a <- data.frame(Samples = sample_names(merged_metagenomes_a),
                   Reads = sample_sums(merged_metagenomes_a))

data_v <- data.frame(Samples = sample_names(merged_metagenomes_v),
                   Reads = sample_sums(merged_metagenomes_v))

# merge kingdom data in to one data frame
data_t <- rbind(data_b, data_e, data_a, data_v)
```

```{r, echo=TRUE, eval=TRUE}
# plot number of reads per kingdom
ggplot(data = data_t, mapping = aes(x = Samples, y = Reads, fill = Samples )) +
  geom_col() +
  theme_classic() +
  ggtitle("Read count per kingdom") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.text.x=element_text(angle=45, hjust=1))
```

```{r}
# reload the biom data, edit column headers and sample name
merged_metagenomes <- import_biom(data)
merged_metagenomes@tax_table@.Data <- substring(merged_metagenomes@tax_table@.Data, 4)
colnames(merged_metagenomes@tax_table@.Data)<- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
colnames(merged_metagenomes@otu_table) <- c("mock1")

# check empty labels (empty = TRUE)
summary(merged_metagenomes@tax_table@.Data== "")
```

```{r}
# transform data frame for plotting
glom <- tax_glom(merged_metagenomes, taxrank = "Species")
mock1_metagenome_species <- psmelt(glom)

# combine genus and species label for scientific naming in plot
mock1_metagenome_species$Species <- as.character(mock1_metagenome_species$Species)
mock1_metagenome_species$Species <- paste(mock1_metagenome_species[,]$Genus,mock1_metagenome_species[,]$Species,  sep=" ", collapse=NULL)
# check label manipulation
# unique(mock1_metagenome_species$Species)
```

```{r}
library(tidyverse)
# alle abundance waarden onder 160.000 opslaan onder één label
mock1_metagenome_species$Species[mock1_metagenome_species$Abundance < 297701] <- "Species < 297701 abund."

mock1_metagenome_species$Species <-  factor(mock1_metagenome_species$Species, levels = unique(mock1_metagenome_species$Species))
```

```{r}
id_species <- mock1_metagenome_species %>% 
  mutate("species_ordered"=reorder(mock1_metagenome_species$Species,mock1_metagenome_species$Abundance)) %>%   
  ggplot(aes(x=species_ordered, y=Abundance))+ 
  geom_col(position = position_dodge())+
  theme(axis.text.x = element_text(angle = 60, hjust = 1), legend.position = "none")+
  labs(
    x = "Species"
  )
id_species
```

```{r}
# transform read count to percentages
glom <- tax_glom(merged_metagenomes, taxrank = "Species")
mock1_metagenome_species_percent <- psmelt(glom)

# normalize read counts to percentages
mock1_metagenome_species_percent$Abundance <- (mock1_metagenome_species_percent$Abundance*100)/sum(mock1_metagenome_species_percent$Abundance)

# combine genus and species label for scientific naming in plot
mock1_metagenome_species_percent$Species <- as.character(mock1_metagenome_species_percent$Species)
mock1_metagenome_species_percent$Species <- paste(mock1_metagenome_species_percent[,]$Genus,mock1_metagenome_species_percent[,]$Species,  sep=" ", collapse=NULL)
mock1_metagenome_species_percent$Species[mock1_metagenome_species_percent$Abundance < 0.5] <- "Species < 0.5% abund."
#unique(mock1_metagenome_species_percent$Species)

# plotten percent
id_species_percent <- mock1_metagenome_species_percent %>% 
  mutate("species_ordered"=reorder(mock1_metagenome_species_percent$Species,mock1_metagenome_species_percent$Abundance)) %>%   
  
  ggplot(aes(x=species_ordered, y=Abundance))+ 
  geom_col(position = position_dodge())+
  theme(axis.text.x = element_text(angle = 60, hjust = 1), legend.position = "none")+
  labs(
    x = "Species"
  )

# id plotten samenvoegen
grid.arrange(id_species, id_species_percent, ncol=2, top=textGrob("Identified species and their corresponding abundance\nread count (right) and percentage (left)\n", gp=gpar(fontsize=15,font=8)))
```

```{r}
# collect composition information
mock1_composition <- as.data.frame(read.csv('/home/daur2/metagenomics/reader_data/HU_waternet_MOCK1_composition.csv', row.names=1, sep = ";"))
```

```{r}
mock1_composition$amount.. <- as.numeric(gsub(",", ".", mock1_composition$amount..))
colnames(mock1_composition) <- c( "name","amount","amountP","sample_name","total_volume")
```

```{r}
# intersect data bases to collect overlap between composition and `kraken2` results
mock1_and_composition_intersect <- mock1_metagenome_species_percent[mock1_metagenome_species_percent$Species %in% mock1_composition$name,]

# collect species that are in composition and not in `kraken2` results
`%!in%` <- Negate(`%in%`) # allow not in = !in
comp_not_in_mock1 <- mock1_composition[mock1_composition$name %!in% mock1_metagenome_species_percent$Species,]

# generate a list of species overlap between mock1 and composition
unique(mock1_and_composition_intersect$Species)
```

```{r}
# generate a list of species that are from composition and not in mock1
unique(comp_not_in_mock1$name)
```

```{r}
# compare labels between `kraken2` subset and composition and store percent values when labels are identical
mock1_and_composition_intersect$amountP <- NA
for (m1_label in mock1_and_composition_intersect$Species){
  for (m1c_label in mock1_composition$name){
    if(m1_label == m1c_label){
      mock1_and_composition_intersect$amountP[mock1_and_composition_intersect$Species == m1_label] <- mock1_composition$amountP[mock1_composition$name == m1c_label]
    }
  }
}
```

```{r}
# load library for melting data frame
library(reshape2)

# collect plotting info and pretify layout format
#colnames(mock1_and_composition_intersect)
mock1_and_comp_plotting_data           <- mock1_and_composition_intersect[,c(10,3,11)]
colnames(mock1_and_comp_plotting_data) <- c("species", "k_abundance", "c_abundance")

mock1_and_comp_plotting_data$species <- reorder(mock1_and_comp_plotting_data$species, mock1_and_comp_plotting_data$k_abundance)

mock1_and_comp_plotting_data       <- melt(mock1_and_comp_plotting_data, id.var = "species")
mock1_and_comp_plotting_data$value <- as.numeric(mock1_and_comp_plotting_data$value)

ggplot(mock1_and_comp_plotting_data, aes(x = species, y = value, fill = variable)) + 
  geom_bar(aes(), stat="identity", position="dodge") +
  theme_classic() +
  ylab("Abundance (%)") +
  xlab("") +
  ggtitle("Abundance comparison between Kraken2 results and composition") +
  theme(plot.title = element_text(hjust = 0.5)) +
  ylim(0,25) +
  theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust = 1)) +
  scale_fill_manual(values=c("skyblue", "orangered"))
```